sanic/sanic.py

```diff
@@ -378,6 +377,63 @@ def stop(self):
             self.sock.close()
         get_event_loop().stop()
 
+    async def create_server(self, host="127.0.0.1", port=8000, debug=False,
+                            before_start=None, after_start=None,
+                            before_stop=None, after_stop=None, ssl=None,
+                            sock=None, loop=None, protocol=HttpProtocol,
+                            backlog=100, stop_event=None):
+        '''
+        Asynchronous version of `run`.
+        '''
+        loop = get_event_loop()
+        server_settings = {
+            'protocol': protocol,
+            'host': host,
+            'port': port,
+            'sock': sock,
+            'ssl': ssl,
+            'debug': debug,
+            'request_handler': self.handle_request,
+            'error_handler': self.error_handler,
+            'request_timeout': self.config.REQUEST_TIMEOUT,
+            'request_max_size': self.config.REQUEST_MAX_SIZE,
+            'loop': loop,
+            'backlog': backlog
+        }
+
+        # -------------------------------------------- #
+        # Register start/stop events
+        # -------------------------------------------- #
+
+        for event_name, settings_name, args, reverse in (
+                ("before_server_start", "before_start", before_start, False),
+                ("after_server_start", "after_start", after_start, False),
+                ("before_server_stop", "before_stop", before_stop, True),
+                ("after_server_stop", "after_stop", after_stop, True),
+                ):
+            listeners = []
+            for blueprint in self.blueprints.values():
+                listeners += blueprint.listeners[event_name]
+            if args:
+                if callable(args):
+                    args = [args]
+                listeners += args
+            if reverse:
+                listeners.reverse()
+            # Prepend sanic to the arguments when listeners are triggered
+            listeners = [partial(listener, self) for listener in listeners]
+            server_settings[settings_name] = listeners
+
+        server_settings['run_async'] = True
+
+        # Serve
+        proto = "http"
+        if ssl is not None:
+            proto = "https"
+        log.info('Goin\' Fast @ {}://{}:{}'.format(proto, host, port))
```

Comment:
@JordanP this introduced an error where the arguments need to be explicitly converted. Please test any code changes if you're unsure, before you make recommendations.